File: 19_数学/中位数与平均数/getMedian.py
from typing import List, Optional


def getMedian(sortedNums: List[int], start: Optional[int] = None, end: Optional[int] = None) -> int:
    """有序数组的中位数(向下取整)."""
    if start is None:
        start = 0
    if end is None:
        end = len(sortedNums)
    if start < 0:
        start = 0
    if end > len(sortedNums):
        end = len(sortedNums)
    if start >= end:
        return 0
    if (end - start) & 1 == 0:
        return (sortedNums[(end + start) // 2 - 1] + sortedNums[(end + start) // 2]) // 2
    return sortedNums[(end + start) // 2]

File: 19_数学/中位数与平均数/getMedian.go
package main

// 求有序数组中位数(向下取整).
func GetMedian(sortedNums []int, start, end int) int {
	if start < 0 {
		start = 0
	}
	if end > len(sortedNums) {
		end = len(sortedNums)
	}
	if start >= end {
		return 0
	}
	if (end-start)&1 == 0 {
		return (sortedNums[(end+start)/2-1] + sortedNums[(end+start)/2]) / 2
	}
	return sortedNums[(end+start)/2]
}
